src/Bynder/Api/Impl/OAuth2/RequestHandler.php

```diff
@@ -7,20 +7,32 @@
 
 class RequestHandler extends AbstractRequestHandler
 {
+    const AUTHORIZATION_CODE = 'authorization_code';
+    const CLIENT_CREDENTIALS = 'client_credentials';
+
     protected $configuration;
 
     private $oauthProvider;
 
+    private $grantType;
+
     public function __construct($configuration)
     {
         $this->configuration = $configuration;
 
+        $redirectUri = $configuration->getRedirectUri();
+
         $this->oauthProvider = new BynderOauthProvider([
             'clientId' => $configuration->getClientId(),
             'clientSecret' => $configuration->getClientSecret(),
-            'redirectUri' => $configuration->getRedirectUri(),
+            'redirectUri' => $redirectUri,
             'bynderDomain' => $configuration->getBynderDomain()
         ]);
+        // Switch between authorization_code and client_credentials based on redirectUri
+        $this->grantType = $redirectUri === null
+            || $configuration->getRedirectUri() === '' ?
+            self::CLIENT_CREDENTIALS :
+            self::AUTHORIZATION_CODE;
```

Comment:
I considered this but in an event someone leaves the `redirectUri` as `' '` in the config file it will make the suggested condition `true` So as a safety check I think we should have both the checks 😄

Reply:
Then still you could do something like `trim($redirectUri) ? self::AUTHORIZATION_CODE : self::CLIENT_CREDENTIALS`.

And besides, in your case it would also be true when it's set to `' '`.

So either way let's turn this into a unit test case :)

Reply:
Sure thing 👍

Reply:
Just coming back to your original reply, my suggestion behaves the same as your current code:

```sh
php > $redirectUri = null;
php > echo !$redirectUri ? 'true' : 'false';
true
php > echo ($redirectUri === '' || $redirectUri === null) ? 'true': 'false';
true
php > $redirectUri = '';
php > echo !$redirectUri ? 'true' : 'false';
true
php > echo ($redirectUri === '' || $redirectUri === null) ? 'true': 'false';
true
php > $redirectUri = ' ';
php > echo !$redirectUri ? 'true' : 'false';
false
php > echo ($redirectUri === '' || $redirectUri === null) ? 'true': 'false';
false
```